core/test_network/src/lib.rs

```diff
@@ -834,20 +841,18 @@ impl TestRuntime for Runtime {
     }
 }
 
-use std::collections::HashSet;
-
-use iroha_config::base::proxy::LoadFromEnv;
-
 impl TestConfiguration for Configuration {
     fn test() -> Self {
-        let mut configuration = iroha::samples::get_config(HashSet::new(), Some(get_key_pair()));
-        configuration
-            .load_environment()
-            .expect("Failed to load configuration from environment");
+        let mut sample_proxy =
+            iroha::samples::get_config_proxy(HashSet::new(), Some(get_key_pair()));
+        let env_proxy = ConfigurationProxy::from_env();
         let (public_key, private_key) = KeyPair::generate().unwrap().into();
-        configuration.public_key = public_key;
-        configuration.private_key = private_key;
-        configuration
+        sample_proxy.public_key = Some(public_key);
+        sample_proxy.private_key = Some(private_key);
+        sample_proxy
+            .combine(env_proxy)
```

Comment:
Shouldn't it be the other way around? Specifically, shouldn't the environment override the sample?

Reply:
That's exactly what was happening, so I'm renaming trait and method to make it more clear as you proposed.